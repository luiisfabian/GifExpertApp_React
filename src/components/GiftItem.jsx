import PropTypes from "prop-types";


export const GiftItem = ({title, urlImg}) => {
  return (
    <div className="card">
        <img src={urlImg} alt={title} />
        <p>{title}</p>
    </div>
  )
}


GiftItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

