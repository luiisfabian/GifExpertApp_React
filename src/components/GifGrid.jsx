import { useState, useEffect } from "react";
import { getGifts } from "../helpers/GetGifs";
import { GiftItem } from "./GiftItem";
import '../styles.css'
import { useEffectGift } from "../hooks/useEffectGift";


export const GifGrid = ({ category }) => {

  const {images, isLoading} = useEffectGift(category);

  return (
    <>
      <h3>{category} </h3>
      {
        isLoading && <h2>Cargando</h2>
      }
      
      <div className="card-grid">

        {
          images.map((image) => {
            // return  <li key={image.id}>{image.title}</li>
            return  <GiftItem key={image.id} {...image}/>


          })
        }
      </div>


    </>
  )
}

