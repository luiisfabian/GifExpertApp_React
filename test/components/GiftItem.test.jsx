import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";

describe('test GiftItemTest', () => {
    const title = "Hola";
    const urlImg = "https://one-punch.com/saitama.jpg";
    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<GiftItem title={title} url={urlImg} />)


        expect(container).toMatchSnapshot();
    })


    test('DEbe mostrar la URl y el Alt indicado', () => {
        render(<GiftItem title={title} urlImg={urlImg} />)
        // expect(screen.getByRole('img').src).toBe(url);

        const { src, alt } = screen.getByRole('img')
        
        expect(src).toBe(urlImg);
        expect(alt).toBe(alt);

    })

    test('Debe Mostrar el titulo en el componente', () => { 

        render(<GiftItem title={title} url={urlImg} />)
        expect(screen.getByText(title)).toBeTruthy();

     })
});