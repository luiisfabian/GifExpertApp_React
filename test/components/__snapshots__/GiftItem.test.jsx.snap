// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GiftItemTest debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
